### PR TITLE
[VECSIM] Use new VecSimQueryResult API

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: src/query.c [hunk 1]
```diff
@@ -1217,9 +1217,12 @@ static sds QueryNode_DumpSds(sds s, const IndexSpec *spec, const QueryNode *qs,
     case QN_VECTOR:
       // TODO: add vector query params
       s = sdscat(s, "VECTOR { ");
-      for (int i = 0; i < qs->vn.vf->resultsLen; i++) {
-        s = sdscatprintf(s, "[%lu,%lf]", qs->vn.vf->results[i].id, qs->vn.vf->results[i].score);
+      VecSimQueryResult_Iterator *iter = VecSimQueryResult_List_GetIterator(qs->vn.vf->results);
+      while (VecSimQueryResult_IteratorHasNext(iter)) {
+        VecSimQueryResult *res = VecSimQueryResult_IteratorNext(iter);
+        s = sdscatprintf(s, "[%lu,%lf]", VecSimQueryResult_GetId(res), VecSimQueryResult_GetScore(res));
       }
```
Comment (rafie) [inline src/query.c:1224]: Maybe:
```
VecSimQueryResult *res = VecSimQueryResult_IteratorNext(iter);
while (res) {
    s = sdscatprintf(s, "[%lu,%lf]", VecSimQueryResult_GetId(res), VecSimQueryResult_GetScore(res));
    res = VecSimQueryResult_IteratorNext(iter);
}
```

File: src/list_reader.c [hunk 2]
```diff
@@ -66,41 +73,43 @@ void ListIterator_Free(struct indexIterator *self) {
...
+  setEof(lr, 1);
 }
 
 static void LR_Rewind(void *ctx) {
   ListIterator *lr = ctx;
-  lr->offset = 0;
+  VecSimQueryResult_IteratorFree(lr->iter);
+  lr->iter = VecSimQueryResult_List_GetIterator(lr->list);
+  lr->lastDocId = 0;
+  setEof(lr, 0);
```
Comment (rafie) [inline src/list_reader.c:99]: Even if it's empty?

File: tests/pytests/test_vecsim.py [hunk 3]
```diff
@@ -98,34 +98,32 @@ def testDel(env):
...
-        env.assertEqual(res[1:3], ['c', ['v_score', '8.30767497366e+34', 'v', 'aacdefgh']])
-        res = env.cmd('FT.SEARCH', 'idx', '@v:[abcdefgh TOPK 2]', 'SORTBY', 'v_score', 'ASC', 'LIMIT', 0, 2)
-        env.assertEqual(res[1:5], ['c', ['v_score', '8.30767497366e+34', 'v', 'aacdefgh'], 'd', ['v_score', '4.78522078483e+37', 'v', 'azcdefgh']])
-        res = env.cmd('FT.SEARCH', 'idx', '@v:[abcdefgh TOPK 3]', 'SORTBY', 'v_score', 'ASC', 'LIMIT', 0, 3)
-        env.assertEqual(res[1:7], ['c', ['v_score', '8.30767497366e+34', 'v', 'aacdefgh'], 'd', ['v_score', '4.78522078483e+37', 'v', 'azcdefgh'],
-                        'b', ['v_score', 'inf', 'v', 'abcdefgg']])
+        res = env.cmd('FT.SEARCH', 'idx', '@v:[aaaaaaaa TOPK 1]', 'SORTBY', 'v_score', 'ASC', 'LIMIT', 0, 1)
+        env.assertEqual(res[1:3], ['b', ['v_score', '3.09485009821e+26', 'v', 'aaaabaaa']])
+        res = env.cmd('FT.SEARCH', 'idx', '@v:[aaaaaaaa TOPK 2]', 'SORTBY', 'v_score', 'ASC', 'LIMIT', 0, 2)
+        env.assertEqual(res[1:5], ['b', ['v_score', '3.09485009821e+26', 'v', 'aaaabaaa'], 'c', ['v_score', '2.02824096037e+31', 'v', 'aaaaabaa']])
```
Comment (alonre24) [inline tests/pytests/test_vecsim.py:126]: If I'm not mistaking, in all of the tests, the results are ordered by their id. Consider adding a test where the expected order of the results when you `SORTBY v_score` is _not_ the ascending oreder of the ids as well (maybe it also related to the bug that Ed found..)

**Review by alonre24 — COMMENTED** [review]
The API changes looks good, added one comment about the tests

File: src/list_reader.c (hunk 2 above)
Comment (ashtul) [reply src/list_reader.c:99]: We will figure it out the first time we run.
This check isn't done on other iterators.

**Pull request merged**